server/pxf-jdbc/src/main/java/org/greenplum/pxf/plugins/jdbc/SQLQueryBuilder.java

```diff
@@ -0,0 +1,309 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.greenplum.pxf.api.BasicFilter;
+import org.greenplum.pxf.api.FilterParser;
+import org.greenplum.pxf.api.io.DataType;
+import org.greenplum.pxf.api.utilities.ColumnDescriptor;
+import org.greenplum.pxf.api.model.RequestContext;
+import org.greenplum.pxf.plugins.jdbc.utils.DbProduct;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+import java.text.ParseException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * SQL query builder.
+ *
+ * Uses {@link JdbcFilterParser} to get array of filters
+ */
+public class SQLQueryBuilder {
+    /**
+     * Construct a new SQLQueryBuilder
+     *
+     * @param context {@link RequestContext}
+     * @param metaData {@link DatabaseMetaData}
+     *
+     * @throws SQLException if some call of DatabaseMetaData method fails
+     */
+    public SQLQueryBuilder(RequestContext context, DatabaseMetaData metaData) throws SQLException {
+        if (context == null) {
+            throw new IllegalArgumentException("Provided RequestContext is null");
+        }
+        requestContext = context;
+        if (metaData == null) {
+            throw new IllegalArgumentException("Provided DatabaseMetaData is null");
+        }
+        databaseMetaData = metaData;
+
+        columns = context.getTupleDescription();
+        tableName = context.getDataSource();
+
+        fillQuoteString();
+    }
+
+    /**
+     * Build SELECT query (with "WHERE" and partition constraints).
+     *
+     * @return Complete SQL query
+     *
+     * @throws ParseException if the constraints passed in RequestContext are incorrect
+     * @throws SQLException if some call of DatabaseMetaData method fails
+     */
+    public String buildSelectQuery() throws ParseException, SQLException {
+        DbProduct dbProduct = DbProduct.getDbProduct(databaseMetaData.getDatabaseProductName());
+
+        StringBuilder sb = new StringBuilder();
+
+        sb.append("SELECT ");
+
+        // Insert columns' names
+        String columnDivisor = "";
+        for (ColumnDescriptor column : columns) {
+            sb.append(columnDivisor);
+            columnDivisor = ", ";
+            sb.append(quoteString + column.columnName() + quoteString);

```

Comment:
Maybe this is an opportunity to start supporting column projection? I think we still need to return the column with null values to the Greenplum side. This doesn't need to go in this PR
```suggestion
            if (column.isProjected()) {
                sb.append(quoteString + column.columnName() + quoteString);
            } else {
                sb.append("null");
            }
```

Reply:
@frankgh -- we will take this as a separate feature after this PR is committed, as it will require an integration test.